src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/TestConnectionManager.kt

```diff
@@ -107,14 +109,726 @@ class TestConnectionManager {
+                                counter.incrementAndGet()
+                            }) {
+                                LOGGER.info { "Canceled $it" }
+                            }
+
+                            Thread.sleep(30000)
+                            LOGGER.info { "Unsubscribing..." }
+                            subscriberMonitor.unsubscribe()
+                        }
+
+                        LOGGER.info { "Sending first message" }
+                        putMessageInQueue(it, queueName)
+
+                        LOGGER.info { "queues list: \n ${getQueuesInfo(it)}" }
+
+                        LOGGER.info { "Sending second message" }
+                        putMessageInQueue(it, queueName)
+                        LOGGER.info { "Sleeping..." }
+                        Thread.sleep(63000)
+
+                        LOGGER.info { "queues list: \n ${getQueuesInfo(it)}" }
+
+
+                        val queuesListExecResult = getQueuesInfo(it)
+                        LOGGER.info { "queues list: \n $queuesListExecResult" }
+
+                        assertEquals(0, getSubscribedChannelsCount(it, queueName))
+                        assertEquals(2, counter.get()) { "Wrong number of received messages" }
+                        assertTrue(
+                            queuesListExecResult.toString().contains("$queueName\t1")
+                        ) { "There should a message left in the queue" }
+                    } finally {
+                        Assertions.assertNotNull(thread)
+                        Assertions.assertDoesNotThrow {
+                            thread!!.interrupt()
+                        }
+                        assertFalse(thread!!.isAlive)
+                    }
                 }
             }
     }
 
+    private fun CountDownLatch.assertComplete(message: String) {
+        assertTrue(
+            await(
+                1L,
+                TimeUnit.SECONDS
+            )
+        ) { "$message, actual count: $count" }
+    }
+
+    private fun <T> assertTarget(target: T, timeout: Long = 1_000, message: String, func: () -> T) {
+        val start = System.currentTimeMillis()
+        while (System.currentTimeMillis() - start < timeout) {
+            if (func() == target) {
+                return
+            }
+            Thread.sleep(100)
+        }
+        assertEquals(target, func(), message)
+    }
+
+    private fun createConnectionManager(container: RabbitMQContainer, configuration: ConnectionManagerConfiguration) =
+        ConnectionManager(
+            RabbitMQConfiguration(
+                host = container.host,
+                vHost = "",
+                port = container.amqpPort,
+                username = container.adminUsername,
+                password = container.adminPassword,
+            ),
+            configuration
+        )
+
     @Test
+    @Disabled
+    // TODO: this test is no more relevant
+    // TODO: we need to change test scenario or remove it
```

Comment:
Could you please elaborate this? What exactly is not longer relevant?

Reply:
This test expected exception, but now channel tries to recover. I've updated test.